lib/PuppeteerSharp/NetworkManager.cs

```diff
@@ -90,18 +96,33 @@ internal Task SetExtraHTTPHeadersAsync(Dictionary<string, string> extraHTTPHeade
 
         internal async Task SetOfflineModeAsync(bool value)
         {
-            if (_offine != value)
-            {
-                _offine = value;
+            _emulatedNetworkConditions.Offline = value;
+            await _updateNetworkConditions().ConfigureAwait(false);
+        }
 
-                await _client.SendAsync("Network.emulateNetworkConditions", new NetworkEmulateNetworkConditionsRequest
-                {
-                    Offline = value,
-                    Latency = 0,
-                    DownloadThroughput = -1,
-                    UploadThroughput = -1
-                }).ConfigureAwait(false);
-            }
+        internal async Task EmulateNetworkConditions(NetworkConditions networkConditions)
+        {
+            _emulatedNetworkConditions.Upload = networkConditions != null
+              ? networkConditions.Upload
+              : -1;
```

Comment:
Can't we do something like ` networkConditions?.Upload ?? -1`

Reply:
I had the same thought but I tried to be faithful to the reference implementation

Reply:
No need here :)

Reply:
If we change it like that, we could also make `NetworkConditions.Upload` nullable.
In the reference implementation, null values are prohibited, so we would break with it.

Reply:
It shouldn't be nullable because we would default it to -1

Reply:
How about we declare it as `public double Upload { get; set; } = -1;` , so it has a more reasonable default value in case the user doesn't assign one.

Reply:
agreed.